File: Daily/PY/Leetcode492-构造矩形.py
from typing import List
from math import sqrt
class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        l,w = 0,0
        for i in range(1,int(sqrt(area) + 1)):
            if area % i != 0: continue
            j = area // i
            if l * w == 0 or l - w > j - i:
                l,w = j,i
        return [l,w]

File: Daily/Rust/Leetcode492-构造矩形.rs
use std::f32;
impl Solution {
    pub fn construct_rectangle(area: i32) -> Vec<i32> {
        let n = f32::sqrt(area as f32) as i32;
        for i in (1..=n).rev() {
            if area % i == 0{
                return vec![area / i,i];
            }
        }
        vec![]
    }
}